docs: Remove stale mention of externalIPs.enabled

nueavv opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: #41043

qmonnet reviewed (changes requested):
Thank you!

Could you please add a bit more context to your commit? Could you have a link to the commit or version notes that removed it in your commit description, for example, please?

We also require that you sign off your commit, as explained in the comment above, please.

nueavv committed:
docs: remove stale mention of externalIPs.enabled

The docs still reference Helm flags that are no longer used in recent
Cilium Helm charts. Since 1.17.6, these options were removed from the chart,
and the kube-proxy replacement features are controlled via
--kube-proxy-replacement=true instead.
https://github.com/cilium/cilium/commit/6a517430c1f945f8d4fdc571c686e41a9dd5bbe2

Per the upgrade guide:
"The flags --enable-node-port (nodePort.enabled in Helm), --enable-host-port,
--enable-external-ips have been deprecated and will be removed in Cilium 1.19.
The kube-proxy replacement features will be only enabled when
--kube-proxy-replacement is set to true."
https://docs.cilium.io/en/stable/operations/upgrade/

This commit updates the docs to remove the stale Helm flags and align the
instructions with current behavior.

Signed-off-by: nueavv <nuguni@kakao.com>

qmonnet reviewed (approved):
This is very complete now, thank you!
